workflow/util/merge.go

```diff
@@ -97,9 +97,36 @@ func JoinWorkflowSpec(wfSpec, wftSpec, wfDefaultSpec *wfv1.WorkflowSpec) (*wfv1.
 	if wfSpec.Suspend != targetWf.Spec.Suspend {
 		targetWf.Spec.Suspend = wfSpec.Suspend
 	}
+	// The value of an argument should respect the priority order of the merge.
+	// However, if Value, ValueFrom and Default are set in different places,
+	// i.e. the Workflow, WorflowTemplate and WorkflowDefault, we should only keep the one with the greatest priority.
+	wfParamsMap := parametersToMapByName(wfSpec)
+	wftParamsMap := parametersToMapByName(wfSpec)
```

Comment:
I think you meant this (though it's kinda a moot point):
```suggestion
	wftParamsMap := parametersToMapByName(wftSpec)
```

Reply:
yes sorry!